honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,309 @@
+
+    Args:
+        vertices: A list of Ladybug Point3D objects.
+
+    Returns:
+         A list of Ladybug Point3D objects with duplicate points removed.
+
+    """
+    return [pt for i, pt in enumerate(vertices)
+            if not pt.is_equivalent(vertices[i - 1], tolerance)]
+
+
+def mesh_to_face3d(mesh):
+    """Create a Ladybug Face3D object out of a rhino3dm Mesh.
+
+    Args:
+        mesh: A rhino3dm mesh geometry.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+
+    faces = []
+    pts = [to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices))]
+
+    for j in range(len(mesh.Faces)):
+        face = mesh.Faces[j]
+        if len(face) == 4:
+            all_verts = (pts[face[0]], pts[face[1]],
+                         pts[face[2]], pts[face[3]])
+        else:
+            all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+
+        faces.append(Face3D(all_verts))
+
+    return faces
+
+
+def extract_mesh_faces_colors(mesh, color_by_face=False):
+    """Extract face indices and colors from a Rhino mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A tuple of mesh faces and mesh colors
+    """
+
+    colors = None
+    lb_faces = []
+    for i in range(len(mesh.Faces)):
+        face = mesh.Faces[i]
+        if len(face) == 4:
+            lb_faces.append((face[0], face[1], face[2], face[3]))
+        else:
+            lb_faces.append((face[0], face[1], face[2]))
+    if len(mesh.VertexColors) != 0:
+        colors = []
+        if color_by_face is True:
+            for j in range(len(mesh.Faces)):
+                face = mesh.Faces[j]
+                col = mesh.VertexColors[face[0]]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+        else:
+            for k in range(len(mesh.VertexColors)):
+                col = mesh.VertexColors[k]
+                colors.append(lbc.Color(col.R, col.G, col.B))
+    return lb_faces, colors
+
+
+def mesh_to_mesh3d(mesh, color_by_face=True):
+    """Ladybug Mesh3D from Rhino Mesh.
+
+    Args:
+        mesh: Rhino3dm mesh
+        color_by_face: Bool. True will derive colors from mesh faces
+
+    Returns:
+        A Ladybug Mesh3D object
+    """
+    lb_verts = tuple(to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices)))
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def brep_to_face3d(brep):
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+
+    return faces
+
+
+def brep2d_to_face3d(brep, tolerance):
```

Comment:
@chriswmackey, can you take a look at this function. I'm not the best person to review this.